File: solve.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

const vector<vector<int>> inputAnswers = {{1, 2, 3, 4, 5}, {1, 3, 2, 4, 2}};
const vector<vector<int>> outputReturn = {{1}, {1, 2, 3}};

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> result(3);
    fill(result.begin(), result.end(), 0);
    vector<int> supoja1 = {1, 2, 3, 4, 5};
    vector<int> supoja2 = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> supoja3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    int max_value = 0;
    for (int i = 0; i < answers.size(); i++){
        if(supoja1[i%supoja1.size()] == answers[i])
            result[0]++;
        if(supoja2[i%supoja2.size()] == answers[i])
            result[1]++;
        if(supoja3[i%supoja3.size()] == answers[i])
            result[2]++;
        max_value = max(max_value, max(result[0], max(result[1], result[2])));
    }
    for (int i = 0; i < 3; i++)
        if(result[i] == max_value)
            answer.push_back(i + 1);
    return answer;
}

int main(){
    for (int i = 0; i < 2; i++){
        const vector<int> result = solution(inputAnswers[i]);
        string output = "O";
        for (int j = 0; j < outputReturn[i].size(); j++){
            if(outputReturn[i][j] != result[j]){
                output = "X";
                break;
            }
        }
        cout << output << "\n";
    }
}

File: solve.js
const input = {
    answers: [
        [1, 2, 3, 4, 5],
        [1, 3, 2, 4, 2],
    ],
    return: [[1], [1, 2, 3]],
};

function solution(answers) {
    const answer = [];
    const result = [0, 0, 0];
    const supoja1 = [1, 2, 3, 4, 5];
    const supoja2 = [2, 1, 2, 3, 2, 4, 2, 5];
    const supoja3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    let max_value = 0;
    for (let i = 0; i < answers.length; i++) {
        if (supoja1[i % supoja1.length] === answers[i]) result[0]++;
        if (supoja2[i % supoja2.length] === answers[i]) result[1]++;
        if (supoja3[i % supoja3.length] === answers[i]) result[2]++;
        max_value = Math.max(max_value, ...result);
    }
    for (let i = 0; i < result.length; i++)
        if (result[i] === max_value) answer.push(i + 1);
    return answer;
}

function test() {
    for (let i = 0; i < 2; i++) {
        const result = solution(input["answers"][i]);
        let ouput = "O";
        for (let j = 0; j < input["return"][i].length; j++) {
            if (input["return"][i][j] !== result[j]) {
                output = "X";
                break;
            }
        }
        console.log(ouput);
    }
}

let start = new Date(); // 시작
test();
let end = new Date(); // 종료

console.log("----------테스트 실행 완료----------");
console.log(`걸린 시간: ${(end - start) / 1000}sec`);
